### PR TITLE
[TODO-List-Cleanup] Delete `NumChildren`, `GetChild` and `gtGetChildPointer`

### PR DESCRIPTION
**Pull request by SingleAccretion**
All these functions represent what is essentially duplicated code.

[No diffs](https://dev.azure.com/dnceng/public/_build/results?buildId=1479506&view=ms.vss-build-web.run-extensions-tab).

### CODE REVIEW

**Comment by SingleAccretion** [comment]
@dotnet/jit-contrib

**Comment by JulieLeeMSFT** [comment]
@AndyAyersMS PTAL.

File: src/coreclr/jit/gentree.cpp [hunk 1]
```diff
@@ -4986,199 +4986,20 @@ unsigned GenTree::GetScaledIndex()
 }
 
 //------------------------------------------------------------------------
-// gtGetChildPointer: If 'parent' is the parent of this node, return the pointer
-//    to the child node so that it can be modified; otherwise, return nullptr.
+// TryGetUse: Get the def -> def edge for a child of this tree.
```
Comment (AndyAyersMS) [inline src/coreclr/jit/gentree.cpp:4989]: Nit: `def` seems like an odd name choice here, and `def -> def` likewise.

Maybe `child` would be better?

**Review by AndyAyersMS — COMMENTED** [review]
Looks good overall, just one nit on parameter naming.

If you change that it, might ripple through to other similar APIs.

File: src/coreclr/jit/gentree.cpp (hunk 1 above)
Comment (SingleAccretion) [reply src/coreclr/jit/gentree.cpp]: Renamed to `operand` as that's what other related APIs use (`ReplaceOperand`, `VisitOperands`, `Operands()`).

**Review by AndyAyersMS — APPROVED** [review]
Thanks, that reads much better.

**Comment by AndyAyersMS** [comment]
Did you look at JIT throughput with this change? Especially for Tier0 code and or Tier0 / w PGO?

Techempower runs show a notable drop in full PGO time to first response on around 11/30-12/1 and from a quick scan, this seems to be the only change that might be relevant. But a drop of this magnitude also seems a bit hard to believe...

image

**Comment by EgorBo** [comment]
@AndyAyersMS I've just checked - it's not this commit (I've checked the commit prior this one)

**Pull request merged**